src/llm/backends/openai.py

```
@@ -109,8 +110,35 @@ def extract_openai_cache_tokens(usage: Any) -> tuple[int, int]:
+        except AuthenticationError as exc:
+            if self._is_nous:
+                logger.warning("Nous API 401 detected — attempting auto-refresh...")
+                from ..nous_refresh import refresh_nous_credentials
+
+                new_key = await refresh_nous_credentials()
+                if new_key:
+                    self._client.api_key = new_key
+                    try:
+                        settings.LLM.NOUS_API_KEY = new_key
+                    except Exception:
+                        pass
```

Ruff (0.15.12)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)